components/brave_wallet/browser/cardano/cardano_wallet_service_unittest.cc

```diff
@@ -228,4 +352,180 @@ TEST_F(CardanoWalletServiceUnitTest, CreateAndSignCardanoTransaction) {
       cardano_test_rpc_server_->captured_raw_tx());
 }
 
+TEST_F(CardanoWalletServiceUnitTest,
+       SignAndPostTransaction_AccountSwitch_TOCTOU) {
```

Comment:
this test kind of feels like junk - I ended up manually verifying this which is where I discovered that the sync calls hold the transactions if multiple are awaiting user interaction. I'll probably drop this unless you see a good reason to keep @supermassive